### PR TITLE
Update proxy-aware samples with new proxy options

### PR DESCRIPTION
**Pull request by bretambrose**
* Update crt to 0.13.7 to pull in some bugfixes
* Update mqtt samples to also support direct mqtt connections via an http proxy

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/mqtt/basic_pub_sub/main.cpp [hunk 1]
```diff
@@ -155,16 +154,16 @@ int main(int argc, char *argv[])
...
+        proxyHost = s_getCmdOption(argv, argv + argc, "--proxy_host");
+    }
 
-        if (s_cmdOptionExists(argv, argv + argc, "--proxy_port"))
-        {
-            proxyPort = static_cast<uint16_t>(atoi(s_getCmdOption(argv, argv + argc, "--proxy_port")));
-        }
+    if (s_cmdOptionExists(argv, argv + argc, "--proxy_port"))
+    {
+        proxyPort = static_cast<uint16_t>(atoi(s_getCmdOption(argv, argv + argc, "--proxy_port")));
```
Comment (justinboswell) [inline samples/mqtt/basic_pub_sub/main.cpp:166]: Do you want to check to see if `atoi` returns -1/0 first?

File: samples/mqtt/raw_pub_sub/main.cpp [hunk 2]
```diff
@@ -124,15 +123,16 @@ int main(int argc, char *argv[])
...
-            proxyPort = static_cast<uint16_t>(atoi(s_getCmdOption(argv, argv + argc, "--proxy_port")));
-        }
+    if (s_cmdOptionExists(argv, argv + argc, "--proxy_host"))
+    {
+        proxyHost = s_getCmdOption(argv, argv + argc, "--proxy_host");
+    }
+
+    if (s_cmdOptionExists(argv, argv + argc, "--proxy_port"))
+    {
+        proxyPort = static_cast<uint16_t>(atoi(s_getCmdOption(argv, argv + argc, "--proxy_port")));
```
Comment (justinboswell) [inline samples/mqtt/raw_pub_sub/main.cpp:135]: Same as above?

**Review by justinboswell — APPROVED** [review]
Fix-or-dont-and-ship!

**Review by graebm — APPROVED** [review]

**Comment by schoy** [comment]
@bretambrose Can you give me some high-level description on how the SDK v2 support HTTP proxy from protocol perspectives? In the previous SDK, the code connected with any targeted endpoint to negotiate TLS and then MQTT protocols with socket in TCP, and typical HTTP proxy doesn't support that, unless the mqtt client support SOCKS (If the proxy also support so). What was the solution to bypass the challenge from HTTP proxy? Thanks.

**Comment by bretambrose** [comment]
Not sure what you mean by challenge in that context.  An http proxy can act as an arbitrary tcp-level proxy in a way not much different than SOCKS.  Our proxy support (in the v2 SDK) can chain to either http or mqtt protocol-wise, but at the C level, it could really go to any protocol: https://github.com/awslabs/aws-c-http/blob/main/include/aws/http/proxy.h#L500 as once the proxy negotiation has finished, you've got a basic io channel that you could attach any kind of protocol handler to.

**Pull request merged**